src/Compilers/CSharp/Portable/CodeGen/Optimizer.cs

```diff
@@ -1571,11 +1571,11 @@ public override BoundNode VisitCatchBlock(BoundCatchBlock node)
             return node.Update(node.Locals, exceptionSourceOpt, exceptionTypeOpt, boundFilter, boundBlock, node.IsSynthesizedAsyncCatchAll);
         }
 
-        public override BoundNode VisitStackAllocArrayCreation(BoundStackAllocArrayCreation node)
+        public override BoundNode VisitConvertedStackAllocExpression(BoundConvertedStackAllocExpression node)
         {
             // CLI spec section 3.47 requires that the stack be empty when localloc occurs.
```

Comment:
Where is this CLI requirement enforced?

Reply:
This expression is never allowed in places where there might be values on the stack (for now). We will look into stack-spilling if that changes in the future.

---
In reply to: [137105053](https://github.com/dotnet/roslyn/pull/21746#discussion_r137105053) [](ancestors = 137105053)